web/interceptor/access_control_test.go

```diff
@@ -124,27 +126,27 @@ func TestAccessControl(t *testing.T) {
 	adminCookie := f(t, admin.ID)
 
 	freeAccessTest := accessTests{
-		{"admin", courseAdminCookie, 0, course.ID, 0, true},
-		{"student", studentCookie, 0, course.ID, 0, true},
-		{"student", groupStudentCookie, 0, course.ID, 0, true},
-		{"user", userCookie, 0, course.ID, 0, true},
-		{"non-teacher admin", adminCookie, 0, course.ID, 0, true},
-		{"empty context", "", 0, course.ID, 0, false},
+		{"admin", courseAdminCookie, 0, course.ID, 0, true, connect.CodePermissionDenied},
```

Comment:
While we are at it, I also suggest that these test tables should use named fields:
```go
	{name: "admin", cookie: courseAdminCookie, userID: 0, courseID: course.ID, groupID: 0, wantAccess: true, wantCode: connect.CodePermissionDenied},
```
For tables for which the `groupID`, `userID` etc isn't relevant, they can be dropped. As in this example.

Moreover, we should also make the tests into a map instead as well. See [here for an example](https://github.com/golang/go/wiki/TableDrivenTests#using-a-map-to-store-test-cases). This means that the `name` field wouldn't be necessary in the struct, but instead be the map key.